fundingmanager_test.go

```diff
@@ -687,16 +711,38 @@ func waitForOpenUpdate(t *testing.T, updateChan chan *lnrpc.OpenStatusUpdate) {
 
 func assertNoChannelState(t *testing.T, alice, bob *testNode,
 	fundingOutPoint *wire.OutPoint) {
-	state, _, err := alice.fundingMgr.getChannelOpeningState(fundingOutPoint)
-	if err != ErrChannelNotFound {
-		t.Fatalf("expected to not find channel state, but got: %v", state)
-	}
 
-	state, _, err = bob.fundingMgr.getChannelOpeningState(fundingOutPoint)
-	if err != ErrChannelNotFound {
-		t.Fatalf("expected to not find channel state, but got: %v", state)
+	assertErrChannelNotFound(t, alice, fundingOutPoint)
+	assertErrChannelNotFound(t, bob, fundingOutPoint)
+}
+
+func assertErrChannelNotFound(t *testing.T, node *testNode,
+	fundingOutPoint *wire.OutPoint) {
+
+	var state channelOpeningState
+	var err error
+	for i := 0; i < 10; i++ {
+		state, _, err = node.fundingMgr.getChannelOpeningState(
+			fundingOutPoint)
+		if err != nil && err != ErrChannelNotFound {
+			t.Fatalf("unable to get channel state: %v", err)
+		}
+
+		if err == ErrChannelNotFound {
+			// Got expected state, return with success.
+			return
+		}
+
+		// Try again in 500 ms.
+		time.Sleep(500 * time.Millisecond)
 	}
 
+	// 10 tries without success.
+	if err != nil {
```

Comment:
I don't think it's possible for `err != nil` given the checks above.

Reply:
Agreed 👍